### PR TITLE
Feature/echo 508 leaf only option

### PR DESCRIPTION
**Pull request by tgoff**

### CODE REVIEW

File: spec/src/features/tree-spec.coffee [hunk 1]
```diff
@@ -282,6 +282,179 @@ describe '"tree" control', ->
...
+
+      $(":jstree li[node_value='level_2_child_1'] > a ").click()
+      expect(form.echoforms('serialize')).not.toMatch(/>top_node_1</)
+      expect(form.echoforms('serialize')).not.toMatch(/>level_1_child_1</)
+      expect(form.echoforms('serialize')).toMatch(/>level_2_child_1</)
+      expect(form.echoforms('serialize')).toMatch(/>level_3_child_1</)
+      expect(form.echoforms('serialize')).toMatch(/>level_3_child_2</)
+      expect(form.echoforms('serialize')).not.toMatch(/>level_2_child_2</)
+
+    it "defaults to true", ->
```
Comment (bilts) [inline spec/src/features/tree-spec.coffee]: Parent contexts should use `describe`, not `it`, `it` is for the actual specs.

File: spec/src/features/tree-spec.coffee [hunk 2]
```diff
@@ -282,6 +282,179 @@ describe '"tree" control', ->
...
+      form = template.form(dom, ui: ui, model: model)
```
Comment (bilts) [inline spec/src/features/tree-spec.coffee]: I assume these three lines should be (and similar elsewhere in this file) should be `beforeEach` blocks.

File: spec/src/features/tree-spec.coffee [hunk 3]
```diff
@@ -282,6 +282,179 @@ describe '"tree" control', ->
...
+
+          it "includes any full subtrees", ->
+            expect(form.echoforms('serialize')).toMatch(/level_2_child_2/)
+
+          it "does not include leaf nodes under full subtrees", ->
+            expect(form.echoforms('serialize')).not.toMatch(/.*required_child/)
+
+
 
   describe "other test cases", ->
```
Comment (bilts) [inline spec/src/features/tree-spec.coffee:477]: I know you didn't change this, but could you? :)

Comment (tgoff) [inline spec/src/features/tree-spec.coffee:477]: which lines are you referring to?

File: src/controls/extras/tree.coffee [hunk 4]
```diff
@@ -79,16 +80,29 @@ class Tree extends Typed
...
-    checked_required_nodes
+
+    #Get all nodes which are required, explicitely checked, or implicitely checked (i.e. all children are checked)
+    #Explicitly checked or imlicitely checked (i.e. all descendants checked, required, or irrelevant)
+    checked = @inputs().find('a.jstree-clicked').parent('[node_value][item-relevant=true][item-required=false]')
+    #Required nodes
+    required = @inputs().find('li[item-required=true][node_value][item-relevant= true]')
+    checked_required_nodes = checked.add(required)
+
+    if @simplify_output
```
Comment (bilts) [inline src/controls/extras/tree.coffee:95]: Does this work bidirectionally? i.e. if I give you a model pre-populated with the simplified output, can you reverse it into tree selections? I'm fairly certain the plugin itself needs that to work, and EDSC definitely depends on it working.

Comment (tgoff) [inline src/controls/extras/tree.coffee:95]: do you mean does specifying a parent node in the initial model cause all its children to be selected?  if so then yes it does.  note that simplified output is not formatted any differently, it is just a subset of the values which would be output without simplified set to true.  The lines you commented on are actually not new functionality, just a refactoring of the existing code.  then I add new logic to select out a subset of those values if simplify is enabled.

Comment (bilts) [inline src/controls/extras/tree.coffee:95]: Great, thanks

**Pull request merged**